bpf/tests/bpf-prog-test-run/bpf-prog-test-run.go

```diff
@@ -0,0 +1,487 @@
+		}).Fatalf("error truncating file: %v", err)
+	}
+	defer outF.Close()
+
+	if _, err = io.Copy(outF, osF); err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnameOut,
+		}).Fatalf("error copying file: %v", err)
+	}
+
+	zero := [4 * 4]byte{}
+	for symName, symOff := range mapSymbols {
+		idOff := symOff + (5 * 4)
+		_, err = outF.WriteAt(zero[:], int64(idOff))
+		if err != nil {
+			log.WithFields(log.Fields{
+				"filename": fnameOut,
+			}).Fatalf("error patching file: %v", err)
+		}
+		log.WithFields(log.Fields{
+			"map": symName,
+		}).Debug("zeroed last fields")
+	}
+
+	return nil
+
+}
+
+func dumpDebugMessages(eventsReader *perf.Reader) error {
+	fmt.Printf("Log Messages:\n")
+	for {
+		record, err := eventsReader.Read()
+		if err != nil {
+			return fmt.Errorf("error reading event buffer: %v\n", err)
+		}
+
+		dm := monitor.DebugMsg{}
+		reader := bytes.NewReader(record.RawSample)
+		if err := binary.Read(reader, byteorder.Native, &dm); err != nil {
+			return fmt.Errorf("Error while parsing debug message: %v\n", err)
+		}
+
+		// special record values set from our bpf programs to mark their termination
+		if dm.SubType == monitor.DbgUnspec && dm.Arg1 == 0xe3d && dm.Arg2 == 0xe3d {
+			break
+		}
+		dm.Dump("")
+	}
+
+	return nil
+}
+
+func testCt4Rst(spec *ebpf.Collection) error {
+
+	eventsReader, err := perf.NewReader(spec.Maps["test_events_map"], 1024*4096)
+	if err != nil {
+		log.WithError(err).Fatal("Cannot initialise BPF perf ring buffer sockets")
+	}
+	defer func() {
+		eventsReader.Close()
+	}()
+
+	packetSyn := gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(packetSyn, gopacket.SerializeOptions{},
+		&layers.Ethernet{
+			SrcMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 10},
+			DstMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 20},
+			EthernetType: layers.EthernetTypeIPv4,
+		},
+		&layers.IPv4{
+			Version:  4,
+			SrcIP:    net.IP{10, 3, 0, 10},
+			DstIP:    net.IP{10, 3, 0, 20},
+			Protocol: layers.IPProtocolTCP,
+			IHL:      5,
+		},
+		&layers.TCP{
+			SrcPort:    3010,
+			DstPort:    3020,
+			DataOffset: 5,
+			SYN:        true,
+		},
+		gopacket.Payload([]byte{'p', 'i', 'z', 'z', 'a', '!', ' ', ':', '-', ')'}),
+	)
+
+	packetRst := gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(packetRst, gopacket.SerializeOptions{},
+		&layers.Ethernet{
+			DstMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 10},
+			SrcMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 20},
+			EthernetType: layers.EthernetTypeIPv4,
+		},
+		&layers.IPv4{
+			Version:  4,
+			DstIP:    net.IP{10, 3, 0, 10},
+			SrcIP:    net.IP{10, 3, 0, 20},
+			Protocol: layers.IPProtocolTCP,
+			IHL:      5,
+		},
+		&layers.TCP{
+			DstPort:    3010,
+			SrcPort:    3020,
+			DataOffset: 5,
+			RST:        true,
+		},
+		gopacket.Payload([]byte{'p', 'i', 'z', 'z', 'a', '!', ' ', ':', '-', ')'}),
+	)
+
+	bpfCtMap := spec.Maps["test_cilium_ct_tcp4_65535"]
+
+	// First Packet: SYN
+	prog1 := spec.Programs["test_ct4_rst1"]
+	if prog1 == nil {
```

Comment:
I think the Go way of writing that would be, but maybe worth double checking:
```suggestion
	if prog1 := spec.Programs["test_ct4_rst1"]; prog1 == nil {
```
Also works for most other checks in this file.

Reply:
`golint` does not complain about it, but  I've seen it complain it when using this syntax in some cases (although don't remember which and it is entirely possible that I'm misremembering) so I tend not to use it.

Reply:
Didn't say it was incorrect Go lint :slightly_smiling_face:
I find it makes the code a bit easier to read, especially when its a successions a checks :wink: 
And we're using it a lot in our code base:
```
$ git grep -P "; \w+ != nil" -- "*.go" | grep -v "^vendor\/" | wc -l
2579
```

Reply:
I personally don't find it more readable but it's probably an acquired taste. I'll change it and repush.

Reply:
It could be worth raising in e.g. our #golang channel to see how folks are usually using it.

Reply:
Pushed a new version with the above syntax. It was not trivial to rebase, so added a new patch on top.